File: code-samples/python/mobile-recharge-plans.py
#!/usr/bin/env python3
"""
KwikAPI SDK — Mobile Recharge Plans
─────────────────────────────────────────────────────────────────────────────
Returns all available prepaid mobile recharge plans for a given operator and telecom circle. Plans are grouped by category (Data, Talktime, SMS, etc.) and include validity, description, and pricing. Use the Operator & Circle Detect API to get the correct opid and state_code for the subscriber's number before calling this.

Endpoint  : POST /api/v2/recharge_plans.php
Group     : Fetching APIs
Rate Limit: Per account

Requirements:
    pip install requests

Environment:
    UAT (testing) : https://uat.kwikapi.com
    Production    : https://www.kwikapi.com  (change BASE_URL)

Get your API key: https://kwikapi.com/auth/register
"""

import requests
from typing import Any

BASE_URL = "https://uat.kwikapi.com"  # Switch to https://www.kwikapi.com for production
API_KEY  = "YOUR_API_KEY"


class KwikAPIError(Exception):
    """Raised when the KwikAPI returns a non-success response."""


def mobile_recharge_plans(api_key: str = 'YOUR_API_KEY', state_code: str = '4', opid: str = '1') -> dict[str, Any]:
    """
    Mobile Recharge Plans

    Args:
        api_key (string): Required. Your KwikAPI API key
        state_code (string): Required. Telecom circle code from Circle Codes API
        opid (int): Required. Operator ID from Biller List API

    Returns:
        dict: Parsed JSON response from KwikAPI.

    Raises:
        KwikAPIError: If the API returns success=false.
        requests.HTTPError: On non-2xx HTTP status.
    """
    url = BASE_URL + "/api/v2/recharge_plans.php"

    with requests.Session() as session:
        session.headers.update({"Accept": "application/json"})
        response = session.post(url, data={'api_key': api_key, 'state_code': state_code, 'opid': opid})

    response.raise_for_status()
    data = response.json()

    if not data.get("success"):
        raise KwikAPIError(data.get("message", "Unknown API error"))

    return data


# ── Example usage ─────────────────────────────────────────────────────────────
if __name__ == "__main__":
    try:
        result = mobile_recharge_plans('YOUR_API_KEY', '4', '1')
        import json
        print(json.dumps(result, indent=2))
    except KwikAPIError as e:
        print(f"API Error: {e}")
        raise SystemExit(1)
    except requests.RequestException as e:
        print(f"Request failed: {e}")
        raise SystemExit(1)

File: code-samples/php/mobile-recharge-plans.php
<?php
/**
 * KwikAPI SDK — Mobile Recharge Plans
 * ─────────────────────────────────────────────────────────────────────────────
 * Returns all available prepaid mobile recharge plans for a given operator and telecom circle. Plans are grouped by category (Data, Talktime, SMS, etc.) and include validity, description, and pricing. Use the Operator & Circle Detect API to get the correct opid and state_code for the subscriber's number before calling this.
 *
 * Endpoint  : POST /api/v2/recharge_plans.php
 * Group     : Fetching APIs
 * Rate Limit: Per account
 *
 * Parameters:
 *   @param  string   $api_key          (required)  Your KwikAPI API key
 *   @param  string   $state_code       (required)  Telecom circle code from Circle Codes API
 *   @param  int      $opid             (required)  Operator ID from Biller List API
 *
 * @return array  Decoded JSON response from KwikAPI
 * @throws RuntimeException  on cURL failure or invalid JSON response
 *
 * Environment:
 *   UAT (testing) : https://uat.kwikapi.com
 *   Production    : https://www.kwikapi.com  (change KWIKAPI_BASE_URL)
 *
 * @link https://kwikapi.com/auth/register  Get your API key
 */

define('KWIKAPI_BASE_URL', 'https://uat.kwikapi.com'); // Switch to https://www.kwikapi.com for production
define('KWIKAPI_KEY', 'YOUR_API_KEY');

/**
 * Mobile Recharge Plans
 *
 *   @param  string   $api_key          (required)  Your KwikAPI API key
 *   @param  string   $state_code       (required)  Telecom circle code from Circle Codes API
 *   @param  int      $opid             (required)  Operator ID from Biller List API
 * @return array
 * @throws RuntimeException
 */
function kwik_mobile_recharge_plans(string $api_key = 'YOUR_API_KEY', string $state_code = '4', string $opid = '1'): array
{
    $ch = curl_init();
    curl_setopt_array($ch, [
        CURLOPT_URL            => 'https://uat.kwikapi.com/api/v2/recharge_plans.php',
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_POST           => true,
        CURLOPT_POSTFIELDS     => http_build_query([
        'api_key' => $api_key,
        'state_code' => $state_code,
        'opid' => $opid,
        ]),
        CURLOPT_SSL_VERIFYPEER => true,
        CURLOPT_HTTPHEADER     => ['Accept: application/json'],
    ]);

    $response = curl_exec($ch);
    $httpCode  = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    $curlError = curl_error($ch);
    curl_close($ch);

    if ($response === false) {
        throw new RuntimeException("cURL error: {$curlError}");
    }

    $data = json_decode($response, true);
    if (json_last_error() !== JSON_ERROR_NONE) {
        throw new RuntimeException('Invalid JSON response (HTTP ' . $httpCode . ')');
    }

    return $data;
}

// ── Example usage ─────────────────────────────────────────────────────────────
try {
    $result = kwik_mobile_recharge_plans('YOUR_API_KEY', 'MH', '1');

    if ($result['success'] ?? false) {
        echo "Success: " . ($result['message'] ?? 'OK') . PHP_EOL;
        echo json_encode($result, JSON_PRETTY_PRINT) . PHP_EOL;
    } else {
        echo "API Error: " . ($result['message'] ?? 'Unknown error') . PHP_EOL;
    }
} catch (RuntimeException $e) {
    echo "Request failed: " . $e->getMessage() . PHP_EOL;
    exit(1);
}
